lib/liquid_pull_to_refresh.dart

```diff
@@ -356,43 +349,54 @@ class LiquidPullToRefreshState extends State<LiquidPullToRefresh>
         // progress ring disappear animation
         _ringDisappearController.animateTo(1.0,
             duration: Duration(
-                milliseconds: widget.springAnimationDurationInMilliseconds),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        widget.animSpeedFactor)
+                    .round()),
             curve: Curves.linear);
 
         // indicator translate out
         _indicatorMoveWithPeakController.animateTo(0.0,
             duration: Duration(
-                milliseconds: widget.springAnimationDurationInMilliseconds),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        widget.animSpeedFactor)
+                    .round()),
             curve: Curves.linear);
         _indicatorTranslateInOutController.animateTo(0.0,
             duration: Duration(
-                milliseconds: widget.springAnimationDurationInMilliseconds),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        widget.animSpeedFactor)
+                    .round()),
             curve: Curves.linear);
 
         //initial value of controller is 1.0
         await _showPeakController.animateTo(0.3,
             duration: Duration(
-                milliseconds:
-                    (widget.springAnimationDurationInMilliseconds).round()),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        widget.animSpeedFactor)
+                    .round()),
             curve: Curves.linear);
 
         _radiusController.animateTo(0.0,
             duration: Duration(
-                milliseconds:
-                    (widget.springAnimationDurationInMilliseconds / 5).round()),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        (widget.animSpeedFactor * 5))
+                    .round()),
             curve: Curves.linear);
 
         _showPeakController.value = 0.175;
         await _showPeakController.animateTo(0.1,
             duration: Duration(
-                milliseconds:
-                    (widget.springAnimationDurationInMilliseconds / 5).round()),
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        (widget.animSpeedFactor * 5))
+                    .round()),
             curve: Curves.easeOut);
         _showPeakController.value = 0.0;
 
         await _positionController.animateTo(0.0,
             duration: Duration(
-                milliseconds: _kIndicatorSnapDuration.inMilliseconds * 2));
+                milliseconds: (widget.springAnimationDurationInMilliseconds /

```

Comment:
@AadumKhor, Did you check that by changing `_kIndicatorSnapDuration` to `springAnimationDurationInMilliseconds`  works fine?

Reply:
Actually both of them work fine. To keep them all in sync I have changed this value. Looked fine on both emulator and profiler.